src/appleseed/renderer/modeling/frame/frame.cpp

```diff
@@ -1010,43 +992,28 @@ namespace
         }
     }
 
-    void transform_to_srgb(Image& image)
+    void transform_to_srgb(Image *image)
     {
-        const CanvasProperties& image_props = image.properties();
+        const CanvasProperties& image_props = image->properties();
 
         for (size_t ty = 0; ty < image_props.m_tile_count_y; ++ty)
         {
             for (size_t tx = 0; tx < image_props.m_tile_count_x; ++tx)
-                transform_to_srgb(image.tile(tx, ty));
+                transform_to_srgb(image->tile(tx, ty));
         }
     }
 
-    void write_png_image(
-        const bf::path&         file_path,
-        const Image&            image,
-        ImageAttributes         image_attributes)
-    {
-        Image transformed_image(image);
-
-        // todo: we may want to be more specific here.
-        if (image.properties().m_channel_count == 4)
-            transform_to_srgb(transformed_image);
-
-        create_parent_directories(file_path);
-
-        const std::string filename = file_path.string();
-        GenericImageFileWriter writer(filename.c_str());
-
-        writer.append_image(&transformed_image);
-
-        writer.set_image_output_format(PixelFormat::PixelFormatUInt8);
-
-        image_attributes.insert("color_space", "sRGB");
-        writer.set_image_attributes(image_attributes);
-
-        writer.write();
-    }
 
+    /*
+     * Export bit depth
+     * OpenEXR .exr        32-bit
```

Comment:
Are you sure we're writing 32-bit OpenEXR files? I'm pretty sure we aren't, we're almost certainly writing 16-bit files instead.

Reply:
Maybe I didn't figure out the bit.
I checked the origin Appleseed's output .exr file, opened it in Photoshop, and I saw
![ps.PNG](https://i.loli.net/2019/03/06/5c7f6b9920094.png).
And it showed the same picture after I changed the code.

Reply:
For the bit, I handle it implicitly, which means OIIO will set the bit for different format automatically.